retry.go

```diff
@@ -390,6 +390,13 @@ func (r *retryHTTP) execute() (res *http.Response, err error) {
 }
 
 func isRetryableError(req *http.Request, res *http.Response, err error) (bool, error) {
+	if err == context.DeadlineExceeded 
+		return false, err
+	} else if err == context.Canceled {
+		// Don't return an error -- the caller canceled the operation
+		return false, nil
```

Comment:
We should definitely return an error here, otherwise the client might think this query ran successful